docs/docs/ProgrammingGuide/run-on-dataproc.md

```diff
@@ -1,52 +1,106 @@
 
----
-## **Before You Start**
+Please disable spark.dynamicAllocation when create cluster.  
+eg. gcloud dataproc clusters create bigdl --project $PROJECT_NAME --worker-machine-type $MACHINETYPE --master-machine-type $MACHINETYPE --num-workers $WORKERNUMBER --properties spark:spark.dynamicAllocation.enabled=false   
+
+ Or please set --conf "spark.dynamicAllocation.enabled=false" when submit spark jobs
 
-Before using BigDL on Dataproc, you need a valid Google Cloud account and setup your Google Cloud SDK (you may refer to [https://cloud.google.com/sdk/docs/how-to](https://cloud.google.com/sdk/docs/how-to) for more instructions).
 
 ---
-## **Create Spark Cluster with BigDL**
+## **Download BigDL**
 
-Run the following command to create your cluster
+BigDL can be downloads from https://bigdl-project.github.io/master/#release-download. It provides prebuild binary for different Spark version. Please ssh on the master VM Instances and download BigDL according to Spark version. 
 ```bash
-gcloud dataproc clusters create bigdl \
-    --initialization-actions gs://dataproc-initial/bigdl.sh \
-    --worker-machine-type n1-highmem-4 \
-    --master-machine-type n1-highmem-2 \
-    --num-workers 2 \
-    --zone us-central1-b \
-    --image-version 1.1
+wget BIGDL_LINK
 ```
-You can change `bigdl` into any other name as the cluster name, and you are also free to upload `scripts/launch-dataproc.sh` into your own Google Cloud Storage bucket and use it instead of `gs://dataproc-initial/bigdl.sh` in the initialization-actions field.
 
-When creating a larger cluster with more workers, it is suggested to pass the number of executor into the script via the metadata field as, 
+After downloaded, you will be able to find a zip file, unzip the file
 ```bash
-gcloud dataproc clusters create bigdl \
-    --initialization-actions gs://dataproc-initial/bigdl.sh \
-    --metadata "NUM_EXECUTORS=8" \
-    --worker-machine-type n1-highmem-4 \
-    --master-machine-type n1-highmem-2 \
-    --num-workers 4 \
-    --num-preemptible-workers 4 \
-    --zone us-central1-b \
-    --image-version 1.1
+unzip xxx.zip
 ```
 
-Please note that it is highly recommended to run BigDL in the region where the compute instances come with Xeon E5 v3 or v4 processors (you may find the [Google Cloud Regions and Zones](https://cloud.google.com/compute/docs/regions-zones/regions-zones) for more details).
-
 ---
-## **Play Around with BigDL**
-Once your dataproc cluster is ready, directly go to the following URL (change `bigdl` into your own cluster name if you are using a different one) to play around BigDL in Jupyter Notebook. Note that you need to [create an SSH tunel and SOCKS proxy](https://cloud.google.com/dataproc/docs/concepts/cluster-web-interfaces) to visit them. 
+## **Run BigDL Scala Examples**
+
+
+You can run BigDL examples using the `run.example.sh` script with following parameters:
```

Comment:
Now you can run BigDL examples on Google Dataproc. For instance, you may use the `run.example.sh` script with following parameters:

Which is run.example.sh located?